tests/e2e/specs/wcpay/shopper/shopper-checkout-purchase-site-editor.spec.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import config from 'config';
+
+const { shopper, merchant } = require( '@woocommerce/e2e-utils' );
+
+/**
+ * Internal dependencies
+ */
+
+import {
+	fillCardDetails,
+	confirmCardAuthentication,
+	setupProductCheckoutNoMiniCart,
+} from '../../../utils/payments';
+
+import { shopperWCP, merchantWCP } from '../../../utils';
+
+const cardTestingPreventionStates = [
+	{ cardTestingPreventionEnabled: false },
+	{ cardTestingPreventionEnabled: true },
+];
+
+describe.each( cardTestingPreventionStates )(
+	'Successful purchase, site builder theme',
+	( { cardTestingPreventionEnabled } ) => {
+		beforeAll( async () => {
+			await merchant.login();
+			await merchantWCP.changeTheme( 'twentytwentyfour' );
```

Comment:
You can make use of the `activateTheme` API provided by `@wordpress/e2e-test-utils` instead of defining a custom method.

```
const { activateTheme } = require( '@wordpress/e2e-test-utils' );
await activateTheme( 'twentytwentyfour' );
```

Reply:
Replaced in https://github.com/Automattic/woocommerce-payments/pull/8164/commits/8a7bbc244edff7f1723e103975ca10733fb3e23b. the suggestion!